contracts/DebtLocker.sol

```diff
@@ -51,8 +60,89 @@ contract DebtLocker is IDebtLocker {
         details_[2] = principalPortion;
     }
 
+    function triggerDefault() external override {
+        require(_isPool(msg.sender), "DL:TD:NOT_PD");
+
+        // Repossess collateral and funds from Loan
+        ( uint256 collateralAssetAmount, ) = IMapleLoanLike(loan).repossess(address(this), address(this));
+
+        address collateralAsset = IMapleLoanLike(loan).collateralAsset();
+        address fundsAsset      = IMapleLoanLike(loan).fundsAsset();
+        
+        if (collateralAsset == fundsAsset) return;
+
+        // Deploy Liquidator contract and transfer collateral
+        liquidator = address(new Liquidator(address(this), collateralAsset, fundsAsset, address(this)));
+        require(ERC20Helper.transfer(collateralAsset, address(liquidator), collateralAssetAmount), "DL:TD:TRANSFER");
+    }
+
+    function setAuctioneer(address auctioneer_) external override {
+        require(_isPoolDelegate(msg.sender), "DL:SA:NOT_PD");
+        Liquidator(liquidator).setAuctioneer(auctioneer_);
+    }
+
+    // TODO: Discuss pros/cons of calculating balance in function instead of using amount param.
+    function pullFunds(address token_, address destination_, uint256 amount_) external override {
+        require(_isPoolDelegate(msg.sender), "DL:PF:NOT_PD");
+        Liquidator(liquidator).pullFunds(token_, destination_, amount_);
+    }
+
+    /**********************/
+    /*** View Functions ***/
+    /**********************/
+
+    function getExpectedAmount(uint256 swapAmount_) external view override returns (uint256 returnAmount_) {
+        address collateralAsset = IMapleLoanLike(loan).collateralAsset();
+        address fundsAsset      = IMapleLoanLike(loan).fundsAsset();
+
+        uint256 oracleAmount = 
+            swapAmount_
+                * IMapleGlobalsLike(_globals()).getLatestPrice(collateralAsset)  // Convert from `fromAsset` value.
+                * 10 ** IERC20Like(fundsAsset).decimals()                        // Convert to `toAsset` decimal precision.
```

Comment:
I'm not a fan of using `decimals` to calculate appropriate amounts, it feels quite error prone to me. Can't we work with absolute amounts? In which currency does `getLatestPrice` returns?

Reply:
Unfortunately we need to, as we need to convert precisions as part of this calculation.